src/percy-agent-client/dom.ts

```diff
@@ -225,6 +226,36 @@ class DOM {
     })
   }
 
+  /**
+   * Capture in-memory canvas elements & serialize them to images into the
+   * cloned DOM.
+   *
+   * Without this, applications that have canvas elements will be missing and
+   * appear broken. The Canvas DOM API allows you to covert them to images, which
+   * is what we're doing here to capture that in-memory state & serialize it
+   * into the DOM Percy captures.
+   *
+   * It's important to note the `.toDataURL` API requires WebGL canvas elements
+   * to use `preserveDrawingBuffer: true`. This is because `.toDataURL` captures
+   * from the drawing buffer, which is cleared after each render by default for
+   * performance.
+   *
+   */
+  private serializeCanvasElements(clonedDOM: HTMLDocument): void {
+    for (const $canvas of this.originalDOM.querySelectorAll('canvas')) {
+      const $image = clonedDOM.createElement('img')
+      const canvasId = $canvas.getAttribute('data-percy-element-id')
+      const $clonedCanvas = clonedDOM.querySelector(`[data-percy-element-id=${canvasId}]`) as any
+
+      $image.setAttribute('style', 'max-width: 100%')
+      $image.classList.add('percy-canvas-image')
+
+      $image.src = $canvas.toDataURL()
+      $image.setAttribute('data-percy-canvas-serialized', 'true')
+      $clonedCanvas.parentElement.appendChild($image)
```

Comment:
If the canvas comes first in a container with siblings this will put the image at the end of the container. We should ensure the correct order by inserting it next to the canvas

```suggestion
      $clonedCanvas.parentElement.insertBefore($image, $clonedCanvas)
```